operator-metadata/automation/tests/test_bundle_automation.py

```diff
@@ -25,34 +33,75 @@ def setUp(self):
                 os.environ[component] = data
 
         self.descriptor = File(TestBundleAutomation.DESCRIPTOR_FILE_PATH)
-        self.csv = File(TestBundleAutomation.CSV_FILE_PATH)
+        self.csv = File(TestBundleAutomation.FRESHMAKER_ENABLED_CSV_FILE_PATH)
 
     def test_collect_component_build_info(self):
         build_info = BundleAutomation.collect_component_build_info()
         self.assertEqual(len(build_info), 5)
 
-    def test_update_csv_file(self):
+    def test_generate_package_name_from_annotation(self):
+        operator_annotation = {"operator-image": "registry.redhat.io/amq-streams/strimzi-rhel9-operator@sha256:95f5aa75cd1f7228e78fd4d88d786713fba4cf828dc22bc2dd1d0380909c1aef"}
+        kafka_prev_annotation = {"kafka-previous-image": "registry.redhat.io/amq-streams/kafka-36-rhel9@sha256:177484ebf6f663eedfc558285e4e89e817277389250aec29323452082b6949e4"}
+        kafka_curr_annotation = {"kafka-current-image": "registry.redhat.io/amq-streams/kafka-37-rhel9@sha256:42bf60ce31540dd61fab2c9886d791e41f063ea6f86628694b9e60e49bc8951b"}
+        bridge_annotation = {"bridge-image": "registry.redhat.io/amq-streams/bridge-rhel9@sha256:4db7b231d68a244f5f71c8587123890e1344be933e1a030097c4b32a9bbfe32d"}
+        maven_annotation =  {"maven-image":  "registry.redhat.io/amq-streams/maven-builder-rhel9@sha256:2393070a07677a482e90f53fef58f12e1c73963cf164738aaa1c095e5fe77d0d"}
+        non_image_annotation = {"test": "test"}
+
+        self.assertEqual(BundleAutomation.generate_package_name_from_annotation(operator_annotation), "amqstreams-operator-container")
+        self.assertEqual(BundleAutomation.generate_package_name_from_annotation(kafka_prev_annotation), "amqstreams-kafka-36-container")
+        self.assertEqual(BundleAutomation.generate_package_name_from_annotation(kafka_curr_annotation), "amqstreams-kafka-37-container")
+        self.assertEqual(BundleAutomation.generate_package_name_from_annotation(bridge_annotation), "amqstreams-bridge-container")
+        self.assertEqual(BundleAutomation.generate_package_name_from_annotation(maven_annotation), "amqstreams-maven-builder-container")
+        self.assertEqual(BundleAutomation.generate_package_name_from_annotation(non_image_annotation), None)
+
+    def test_update_freshmaker_disabled_csv_file(self):
```

Comment:
Ok this could be me missing context from the process, but I can't tell from reading the test what its actually demonstrating. Questions I currently have

1. Whats the relationship between freshmaker disabled and the number of replacements?
2. Why are the pull specs all SHA's in the disabled case? 
3. Are we testing multiple things at once? Freshmaker disabled and sha based pull specs.
4. There are no repos in the pull specs, is that a valid pull spec or test?
5. Is freshmaker status supposed to drive the expect_related_images flag? 

I'm sure the test helped you here and is demonstrating something valid but as a casual reader of the code its not telling *me* anything useful. Tests are an executable specification of the behaviour so its really helpful when refactoring to understand what the existing test suite is trying to prove.

Reply:
Refactored the tests to make things more clear, basically we are just testing the old and new cluster service version formats.

> Whats the relationship between freshmaker disabled and the number of replacements?
> Why are the pull specs all SHA's in the disabled case?
> Are we testing multiple things at once? Freshmaker disabled and sha based pull specs.

This was a misnomer in the code, it's not about freshmaker-enabled and freshmaker-disabled format but more about a new format that is cleaner and works better with OSBS automation. Refactored the code to fixthis misnomer and make it more clear

> There are no repos in the pull specs, is that a valid pull spec or test?

This is somewhat of a misnomer as well. It's not really a map of pull_specs but values used to replace parts of the pull_specs themselves. Refactored just now to make it more clear.

>  freshmaker status supposed to drive the expect_related_images flag?

OSBS will